### PR TITLE
add code for to_numeric

### LINKED ISSUE #26: to_numeric

### LINKED ISSUE #30: to_numeric tests

### LINKED ISSUE #37: Add to_numeric example to readme

### PR DESCRIPTION
**Pull request by GuanshuTao**
Closes #26 
Closes #30 
Closes #37

### CODE REVIEW

File: nicenumber/nicenumber.py [hunk 1]
```diff
@@ -123,8 +127,37 @@ def to_numeric(string:str, family:str = 'number'):
     >>> to_numeric(1.5M)
     1500000
     """
+    def raise_err(err):
```
Comment (wiwang) [inline nicenumber/nicenumber.py:130]: move this function outside as a common function

**Review by wiwang — CHANGES_REQUESTED** [review]
Just make changes as we discussed in meeting

File: nicenumber/nicenumber.py (hunk 1 above)
Comment (GuanshuTao) [reply nicenumber/nicenumber.py]: Moved in this pr

**Review by wiwang — APPROVED** [review]

File: nicenumber/nicenumber.py [hunk 2]
```diff
@@ -104,27 +149,80 @@ def raise_err(err):
...
+
+    if not suff in suffix_list: 
+        err = ValueError(
+            f'Invalid string suffix: "{suff}". Valid options: {suffix_list}')
+        return raise_err(err, errors)
+
+    power = suffix_list.index(suff)
+
+    # extract number from string
+    # pattern = digit one or more times, decimal zero or more, digit one or more
```
Comment (jaymegordo) [inline nicenumber/nicenumber.py:223]:
```suggestion
    # pattern = digit one or more times, decimal zero or more, digit zero or more
```

**Review by jaymegordo — APPROVED** [review]

**Pull request merged**